Content.Server/Speech/Components/ListenWireAction.cs

```diff
@@ -0,0 +1,116 @@
+
+public sealed partial class ListenWireAction : BaseToggleWireAction
+{
+    private WiresSystem _wires = default!;
+
+    /// <summary>
+    /// Length of the gibberish string sent when pulsing the wire
+    /// </summary>
+    private int _noiseLength = 16;
+    public override Color Color { get; set; } = Color.Green;
+    public override string Name { get; set; } = "wire-name-listen";
+
+    public override object? StatusKey { get; } = ListenWireActionKey.StatusKey;
+
+    public override object? TimeoutKey { get; } = ListenWireActionKey.TimeoutKey;
+
+    public override int Delay { get; } = 10;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _wires = EntityManager.System<WiresSystem>();
+    }
+    public override StatusLightState? GetLightState(Wire wire)
+    {
+        if (GetValue(wire.Owner))
+            return StatusLightState.On;
+        else
+        {
+            if (TimeoutKey != null && _wires.HasData(wire.Owner, TimeoutKey))
+                return StatusLightState.BlinkingSlow;
+            return StatusLightState.Off;
+        }
+    }
+    public override void ToggleValue(EntityUid owner, bool setting)
+    {
+        if (setting)
+        {
+            // If we defer removal, the status light gets out of sync
+            EntityManager.RemoveComponent<BlockListeningComponent>(owner);
+        }
+        else
+        {
+            EntityManager.EnsureComponent<BlockListeningComponent>(owner, out _);
+        }
+    }
+
+    public override bool GetValue(EntityUid owner)
+    {
+        return !EntityManager.HasComponent<BlockListeningComponent>(owner);
+    }
+
+    public override void Pulse(EntityUid user, Wire wire)
+    {
+        if (!GetValue(wire.Owner) || !IsPowered(wire.Owner))
+            return;
+
+        base.Pulse(user, wire);
+
+        // We have to use a valid euid in the ListenEvent. The user seems
+        // like a sensible choice, but we need to mask their name.
+
+        // Save the user's existing voicemask if they have one
+        bool oldEnabled = true;
+        string oldVoiceName = "ERROR";
+        if (EntityManager.TryGetComponent<VoiceMaskComponent>(user, out var oldMask))
+        {
+            oldEnabled = oldMask.Enabled;
+            oldVoiceName = oldMask.VoiceName;
+        }
+
+        // Give the user a temporary voicemask component
+        var mask = EntityManager.EnsureComponent<VoiceMaskComponent>(user);
+        mask.Enabled = true;
+        mask.VoiceName = Loc.GetString("wire-listen-pulse-identifier");
+
+        var chars = Loc.GetString("wire-listen-pulse-characters").ToCharArray();
+        var noiseMsg = BuildGibberishString(chars, _noiseLength);
+
+        // Send as a ListenEvent to bypass getting blocked by ListenAttemptEvent
+        var ev = new ListenEvent(noiseMsg, user);
+        EntityManager.EventBus.RaiseLocalEvent(wire.Owner, ev);
+
+        // Remove the voicemask component, or set it back to what it was before
```

Comment:
This should call a method directly rather than relying on an event to bypass something because if more functionality gets added it just ignores it.

Reply:
I'm pretty reluctant to use a direct call, since that would mean coupling it pretty tightly with RadioDeviceSystem, and there could (theoretically?) be use for mic wires on non-radio devices.

I think I can make a small change and remove the need for the bypass though. That way it will respect future additions.

Reply:
Moved the call to base.Pulse to the end of the method, so a ListenAttemptEvent can go out and pass/fail normally before adding the BlockListeningComponent. Now anything else that could block the listen attempt will be respected.